huaweicloud/services/acceptance/workspace/data_source_huaweicloud_workspace_tags_test.go

```diff
@@ -0,0 +1,175 @@
+package workspace
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccWorkspaceTagsDataSource_basic(t *testing.T) {
+	var (
+		rName = acceptance.RandomAccResourceNameWithDash()
+
+		desktop      interface{}
+		resourceName = "huaweicloud_workspace_desktop.test"
+		rc           = acceptance.InitResourceCheck(
+			resourceName,
+			&desktop,
+			getDesktopFunc,
+		)
+
+		dcName = "data.huaweicloud_workspace_tags.all"
+		dc     = acceptance.InitDataSourceCheck(dcName)
+
+		dcByKeyName = "data.huaweicloud_workspace_tags.filter_by_key"
+		dcByKey     = acceptance.InitDataSourceCheck(dcByKeyName)
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
```

Comment:
```suggestion
		CheckDestroy:      rc.CheckResourceDestroy(),
```
delete CheckDestroy,  CheckDestroy only applies to resources